src/PowerShell/PowerShellExtensions.cs

```diff
@@ -37,5 +37,13 @@ public static Collection<T> InvokeAndClearCommands<T>(this PowerShell pwsh)
                 pwsh.Commands.Clear();
             }
         }
+
+        public static string FormatObjectToString(this PowerShell pwsh, object inputObject)
+        {
+            return pwsh.AddCommand("Write-Output")
+                .AddParameter("InputObject", inputObject)
+                .AddCommand("Out-String")
+                .InvokeAndClearCommands<string>()[0];
```

Comment:
It looks to me we can depends on the overriden method `ToString()` of `PSObject` to do the same. See here:

https://github.com/PowerShell/PowerShell/blob/655b8b614c7dbb824b3e16f9968b73c38e8a023e/src/System.Management.Automation/engine/MshObject.cs#L1426-L1435

Reply:
Talked to Tyler offline. It turns out `ToString(ExecutionContext context, ...)` doesn't do what I thought it would do. Close this comment.